src/extension.ts

```diff
@@ -83,12 +84,30 @@ export function activate(context: ExtensionContext) {
 
     executeAmebaOnWorkspace(ameba);
 
+    // This can happen when a file is open _or_ when a file's language id changes
     workspace.onDidOpenTextDocument(doc => {
-        ameba && ameba.execute(doc);
+        if (ameba && ameba.config.trigger !== LintTrigger.None && isCrystalDocument(doc)) {
+            if (isDocumentVirtual(doc)) {
+                if (ameba.config.trigger === LintTrigger.Type) {
+                    outputChannel.appendLine(`[Open] Running ameba on ${getRelativePath(doc)}`);
+                    ameba.execute(doc, true);
+                }
+            } else {
+                outputChannel.appendLine(`[Open] Running ameba on ${getRelativePath(doc)}`);
+                ameba.execute(doc);
+            }
```

Comment:
Given that the document was just opened, it can't be virtual, no?

```suggestion
            outputChannel.appendLine(`[Open] Running ameba on ${getRelativePath(doc)}`);
            ameba.execute(doc);
```

Reply:
This callback is also called when any text document or virtual/untitled is changed to a different language, meaning we should handle that edge case